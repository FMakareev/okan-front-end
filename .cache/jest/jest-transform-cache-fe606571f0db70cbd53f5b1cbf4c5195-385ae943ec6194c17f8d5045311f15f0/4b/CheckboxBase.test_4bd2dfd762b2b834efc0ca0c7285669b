c86c0f7a1eca1d1059ee9009ed40ab6d
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

require("jest-styled-components");

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _CheckboxBase = require("./CheckboxBase");

var _StyledThemeProvider = require("../../styles/StyledThemeProvider");

var _onCheckbox = _interopRequireDefault(require("../../assets/image/onCheckbox.png"));

var _offCheckbox = _interopRequireDefault(require("../../assets/image/offCheckbox.png"));

test('Disabled checkbox', function () {
  var tree = _reactTestRenderer["default"].create(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_CheckboxBase.CheckboxBase, {
    index: 1
  }, "CheckboxBase"))).toJSON();

  expect(tree).toMatchSnapshot();
});
test('Enabled checkbox', function () {
  var tree = _reactTestRenderer["default"].create(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_CheckboxBase.CheckboxBase, {
    checked: true,
    index: 1
  }, "CheckboxBase"))).toJSON();

  expect(tree).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkNoZWNrYm94QmFzZS50ZXN0LmpzIl0sIm5hbWVzIjpbInRlc3QiLCJ0cmVlIiwicmVuZGVyZXIiLCJjcmVhdGUiLCJ0b0pTT04iLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiXSwibWFwcGluZ3MiOiI7Ozs7QUFBQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFDQTs7QUFFQTs7QUFDQTs7QUFFQUEsSUFBSSxDQUFDLG1CQUFELEVBQXNCLFlBQU07QUFDOUIsTUFBTUMsSUFBSSxHQUFHQyw4QkFDVkMsTUFEVSxDQUVULGdDQUFDLHdDQUFELFFBQ0UsZ0NBQUMsMEJBQUQ7QUFBYyxJQUFBLEtBQUssRUFBRTtBQUFyQixvQkFERixDQUZTLEVBTVZDLE1BTlUsRUFBYjs7QUFPQUMsRUFBQUEsTUFBTSxDQUFDSixJQUFELENBQU4sQ0FBYUssZUFBYjtBQUNELENBVEcsQ0FBSjtBQVdBTixJQUFJLENBQUMsa0JBQUQsRUFBcUIsWUFBTTtBQUM3QixNQUFNQyxJQUFJLEdBQUdDLDhCQUNWQyxNQURVLENBRVQsZ0NBQUMsd0NBQUQsUUFDRSxnQ0FBQywwQkFBRDtBQUFjLElBQUEsT0FBTyxNQUFyQjtBQUFzQixJQUFBLEtBQUssRUFBRTtBQUE3QixvQkFERixDQUZTLEVBUVZDLE1BUlUsRUFBYjs7QUFTQUMsRUFBQUEsTUFBTSxDQUFDSixJQUFELENBQU4sQ0FBYUssZUFBYjtBQUNELENBWEcsQ0FBSiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XHJcbmltcG9ydCAnamVzdC1zdHlsZWQtY29tcG9uZW50cyc7XHJcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcclxuaW1wb3J0IHsgQ2hlY2tib3hCYXNlIH0gZnJvbSAnLi9DaGVja2JveEJhc2UnO1xyXG5pbXBvcnQgeyBTdHlsZWRUaGVtZVByb3ZpZGVyIH0gZnJvbSAnLi4vLi4vc3R5bGVzL1N0eWxlZFRoZW1lUHJvdmlkZXInO1xyXG5cclxuaW1wb3J0IG9uQ2hlY2tib3ggZnJvbSAnLi4vLi4vYXNzZXRzL2ltYWdlL29uQ2hlY2tib3gucG5nJztcclxuaW1wb3J0IG9mZkNoZWNrYm94IGZyb20gJy4uLy4uL2Fzc2V0cy9pbWFnZS9vZmZDaGVja2JveC5wbmcnO1xyXG5cclxudGVzdCgnRGlzYWJsZWQgY2hlY2tib3gnLCAoKSA9PiB7XHJcbiAgY29uc3QgdHJlZSA9IHJlbmRlcmVyXHJcbiAgICAuY3JlYXRlKFxyXG4gICAgICA8U3R5bGVkVGhlbWVQcm92aWRlcj5cclxuICAgICAgICA8Q2hlY2tib3hCYXNlIGluZGV4PXsxfT5DaGVja2JveEJhc2U8L0NoZWNrYm94QmFzZT5cclxuICAgICAgPC9TdHlsZWRUaGVtZVByb3ZpZGVyPixcclxuICAgIClcclxuICAgIC50b0pTT04oKTtcclxuICBleHBlY3QodHJlZSkudG9NYXRjaFNuYXBzaG90KCk7XHJcbn0pO1xyXG5cclxudGVzdCgnRW5hYmxlZCBjaGVja2JveCcsICgpID0+IHtcclxuICBjb25zdCB0cmVlID0gcmVuZGVyZXJcclxuICAgIC5jcmVhdGUoXHJcbiAgICAgIDxTdHlsZWRUaGVtZVByb3ZpZGVyPlxyXG4gICAgICAgIDxDaGVja2JveEJhc2UgY2hlY2tlZCBpbmRleD17MX0+XHJcbiAgICAgICAgICBDaGVja2JveEJhc2VcclxuICAgICAgICA8L0NoZWNrYm94QmFzZT5cclxuICAgICAgPC9TdHlsZWRUaGVtZVByb3ZpZGVyPixcclxuICAgIClcclxuICAgIC50b0pTT04oKTtcclxuICBleHBlY3QodHJlZSkudG9NYXRjaFNuYXBzaG90KCk7XHJcbn0pO1xyXG4iXX0=