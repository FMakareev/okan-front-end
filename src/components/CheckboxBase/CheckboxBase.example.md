CheckboxBase example:

```jsx harmony
<CheckboxBase />
```

CheckboxBase example:
```jsx harmony
<CheckboxBase index={1}>
Hello
</CheckboxBase>
```
CheckboxBase example:
```jsx harmony
<CheckboxBase disabled>
Hello
</CheckboxBase>
```

CheckboxBase example:
```jsx harmony
<CheckboxBase checked index={2}>
Hello
</CheckboxBase>
```
