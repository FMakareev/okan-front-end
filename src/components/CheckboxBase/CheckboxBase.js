import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { space } from 'styled-system';

/** Image */
import onCheckbox from '../../assets/image/onCheckbox.png';
import offCheckbox from '../../assets/image/offCheckbox.png';

const Wrapper = styled.div`
  ${space};
`;

const Input = styled.input`
  position: absolute;
  opacity: 0;

  & + label {
    position: relative;
    cursor: pointer;
    padding: 0;
  }

  & + label:before {
    content: '';
    position: absolute;
    left: 0px;
    top: 0px;
    width: 32px;
    height: 32px;
    background-image: url(${offCheckbox}) !important;
    background-repeat: no-repeat;
    box-sizing: border-box;
  }

  &:checked + label:before {
    display: none;
  }

  &:disabled + label {
    color: #b8b8b8;
    cursor: auto;
  }

  &:disabled + label:before {
    box-shadow: none;
    background-color: #ddd;
  }

  &:checked + label:after {
    content: '';
    position: absolute;
    left: 0px;
    top: -2px;
    width: 32px;
    height: 32px;
    background-image: url(${onCheckbox}) !important;
    background-repeat: no-repeat;
  }
`;

/**
 * Компонент чекбокса базового
 * @example ./CheckboxBase.example.md
 */
export class CheckboxBase extends Component {
  static propTypes = {
    // input: inputPropTypes,
    /** The disabled attribute specifies that the input field is disabled. */
    disabled: PropTypes.bool,
    /** Text field form with a pre-selected checkbox. */
    checked: PropTypes.bool,
    /** . */
    index: PropTypes.number,
  };

  static defaultProps = {};

  render() {
    const { input, index, checked, disabled, children, onClick } = this.props;

    return (
      <Wrapper>
        <Input
          id={`styled-checkbox-${index || input.name}`}
          type="checkbox"
          checked={input ? input.value : false}
          disabled={disabled}
          onClick={onClick}
          {...input}
        />

        <label htmlFor={`styled-checkbox-${index || input.name}`}>{children}</label>
      </Wrapper>
    );
  }
}

export default CheckboxBase;
