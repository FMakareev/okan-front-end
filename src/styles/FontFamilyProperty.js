/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
