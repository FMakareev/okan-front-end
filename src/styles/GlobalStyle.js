import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
