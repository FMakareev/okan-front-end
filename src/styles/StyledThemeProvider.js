import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
