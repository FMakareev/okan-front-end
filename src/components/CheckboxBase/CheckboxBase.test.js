import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { CheckboxBase } from './CheckboxBase';
import { StyledThemeProvider } from '../../styles/StyledThemeProvider';

test('Disabled checkbox', () => {
  const tree = renderer
    .create(
      <StyledThemeProvider>
        <CheckboxBase index={1}>CheckboxBase</CheckboxBase>
      </StyledThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Enabled checkbox', () => {
  const tree = renderer
    .create(
      <StyledThemeProvider>
        <CheckboxBase checked index={1}>
          CheckboxBase
        </CheckboxBase>
      </StyledThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
