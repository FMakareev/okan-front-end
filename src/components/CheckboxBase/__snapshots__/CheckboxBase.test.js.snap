// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled checkbox 1`] = `
.c1 {
  position: absolute;
  opacity: 0;
}

.c1 + label {
  position: relative;
  display: block;
  padding: 0;
  cursor: pointer;
}

.c1 + label:before {
  content: '';
  position: absolute;
  left: 0px;
  top: 0px;
  width: 32px;
  height: 32px;
  background-image: url(test-file-stub) !important;
  background-repeat: no-repeat;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.c1:checked + label:before {
  display: none;
}

.c1:disabled + label {
  color: #b8b8b8;
  cursor: auto;
}

.c1:disabled + label:before {
  -webkit-box-shadow: none;
  box-shadow: none;
  background-color: #ddd;
}

.c1:checked + label:after {
  content: '';
  position: absolute;
  left: 0px;
  top: -2px;
  width: 32px;
  height: 32px;
  background-image: url(test-file-stub) !important;
  background-repeat: no-repeat;
}

.c0 {
  font-family: system-ui,sans-serif;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className=""
  >
    <input
      className="c1"
      id="styled-checkbox-undefined"
      name="partnersundefined"
      target="div"
      toString={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="styled-checkbox-undefined"
    >
      CheckboxBase
    </label>
  </div>
</div>
`;

exports[`Enabled checkbox 1`] = `
.c1 {
  position: absolute;
  opacity: 0;
}

.c1 + label {
  position: relative;
  display: block;
  padding: 0;
  cursor: pointer;
}

.c1 + label:before {
  content: '';
  position: absolute;
  left: 0px;
  top: 0px;
  width: 32px;
  height: 32px;
  background-image: url(test-file-stub) !important;
  background-repeat: no-repeat;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.c1:checked + label:before {
  display: none;
}

.c1:disabled + label {
  color: #b8b8b8;
  cursor: auto;
}

.c1:disabled + label:before {
  -webkit-box-shadow: none;
  box-shadow: none;
  background-color: #ddd;
}

.c1:checked + label:after {
  content: '';
  position: absolute;
  left: 0px;
  top: -2px;
  width: 32px;
  height: 32px;
  background-image: url(test-file-stub) !important;
  background-repeat: no-repeat;
}

.c0 {
  font-family: system-ui,sans-serif;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className=""
  >
    <input
      checked={true}
      className="c1"
      id="styled-checkbox-undefined"
      name="partnersundefined"
      target="div"
      toString={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="styled-checkbox-undefined"
    >
      CheckboxBase
    </label>
  </div>
</div>
`;
